src/literal.h

```diff
@@ -71,9 +71,13 @@ class Literal {
   int64_t getInteger() const;
   double getFloat() const;
   int64_t getBits() const;
+  // Equality checks for the type and the bits, so a nan float would
+  // be compared bitwise.
   bool operator==(const Literal& other) const;
   bool operator!=(const Literal& other) const;
-  bool bitwiseEqual(const Literal& other) const;
+  // Compare in a non-bitwise manner, which means for a nan a we have
+  // a != a
+  bool nonBitwiseEqual(const Literal& other) const;
```

Comment:
This is just a little ambiguous whether the 'non' means 'not bitwise' or 'not equal'. Maybe 'typedEqual' or something like that would indicate what it is, rather than what it isn't?